python/benchmark/gen_data_distributed.py

```diff
@@ -204,26 +204,42 @@ def gen_dataframe(self, spark: SparkSession) -> Tuple[DataFrame, List[str]]:
         tail = tail_strength * np.exp(-0.1 * singular_ind / effective_rank)
         # S and V are generated upfront, U is generated across partitions.
         s = np.identity(n) * (low_rank + tail)
-        v, _ = linalg.qr(
-            generator.standard_normal(size=(cols, n)),
-            mode="economic",
-            check_finite=False,
+        v, _ = np.linalg.qr(
+            np.random.standard_normal(size=(cols, n)),
+            mode="reduced",
         )
 
         # Precompute the S*V.T multiplicland with partition-wise normalization.
         sv_normed = np.dot(s, v.T) * np.sqrt(1 / num_partitions)
+        del s
+        del v
+
+        maxRecordsPerBatch = int(
+            spark.sparkContext.getConf().get(
+                "spark.sql.execution.arrow.maxRecordsPerBatch", "10000"
+            )
+        )
 
         # UDF for distributed generation of U and the resultant product U*S*V.T
         def make_matrix_udf(iter: Iterable[pd.DataFrame]) -> Iterable[pd.DataFrame]:
             for pdf in iter:
+                import cupy as cp
+
                 partition_index = pdf.iloc[0][0]
                 n_partition_rows = partition_sizes[partition_index]
-                u, _ = linalg.qr(
-                    generator.standard_normal(size=(n_partition_rows, n)),
-                    mode="economic",
-                    check_finite=False,
+                u, _ = cp.linalg.qr(
+                    cp.random.standard_normal(size=(n_partition_rows, n)),
+                    mode="reduced",
                 )
-                yield pd.DataFrame(data=np.dot(u, sv_normed))
+                data = cp.dot(u, cp.asarray(sv_normed))
+                del u
+                for i in range(0, n_partition_rows, maxRecordsPerBatch):
```

Comment:
This part will be common to the different generators, I think, so for next one, would be good to find a way to share.